client/dashboard/sites/staging-site-sync-dropdown/index.tsx

```diff
@@ -42,6 +44,19 @@ export default function StagingSiteSyncDropdown( {
 	const productionSiteId = site ? getProductionSiteId( site ) : null;
 	const stagingSiteId = site ? getStagingSiteId( site ) : null;
 
+	// Check atomic transfer status for staging site
+	const { data: atomicTransfer } = useQuery( {
+		...siteLatestAtomicTransferQuery( stagingSiteId ?? 0 ),
+		refetchInterval: ( query ) => {
+			return isAtomicTransferInProgress( query.state.data?.status ?? 'pending' ) ? 3000 : false;
+		},
+		enabled: !! stagingSiteId,
+	} );
```

Comment:
I wonder if we could reuse this part, i.e. not having this code duplicated across both files.